src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,166 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * TaskList Class is responsible for housing the taskList array and methods for adding and removing tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList = new ArrayList<>();
+    private ArrayList<Task> findList = new ArrayList<>();
+
+    /**
+     * Empty constructor for TaskList.
+     */
+    public TaskList() {
+    }
+
+    /**
+     * Marks a mask as complete.
+     * @param index
+     */
+    public String markTask(int index) {
+        Task currentTask = taskList.get(index);
+        String reply = "";
+        reply += "We have completed this task!\n";
+        currentTask.mark();
+        reply += currentTask.getTaskType() + " " + currentTask.getStatus()
+                + " " + currentTask.getTask();
+
+        return reply;
+    }
+
+    /**
+     * Deletes a task from the TaskList.
+     * @param index
+     */
+    public String deleteTask(int index) {
+        Task currentTask = taskList.get(index);
+        String reply = "";
+
+        reply += "Task has been deleted!\n";
+        reply += currentTask.getTaskType() + " " + currentTask.getStatus()
+                + " " + currentTask.getTask() + "\n";
+
+        taskList.remove(index);
+        return reply;
+    }
+
+    /**
+     * Unmarks a task as completed.
+     * @param index
+     */
+    public String unmarkTask(int index) {
+        Task currentTask = taskList.get(index);
+        String reply = "";
+        reply += "Oops, task unmarked!\n";
+        currentTask.unmark();
+        reply += currentTask.getTaskType() + " " + currentTask.getStatus()
+                + " " + currentTask.getTask() + "\n";
+
+        return reply;
+    }
+
+    /**
+     * Lists all the tasks currently created.
+     */
+    public String listTask() {
+        String reply = "";
+        reply += "These are the tasks we currently have: \n";
+
+        for (int i = 0; i < taskList.size(); i++) {
+            Task currentTask = taskList.get(i);
+            reply += (i + 1) + ". " + currentTask.toString() + "\n";
+        }
+
+        reply += "We have " + (taskList.size()) + " tasks.\n";
+
+        return reply;
+    }
+
+    /**
+     * Adds new task to the TaskListt.
+     * @param first
+     * @param second

```

Comment:
The parameter names are a bit confusing - perhaps you may want to specify that 'first' is the Task type and second is the rest of the String?